code/game/world.dm

```diff
@@ -10,6 +10,31 @@
 GLOBAL_DATUM_INIT(init, /datum/global_init, new)
 GLOBAL_DATUM(config, /datum/configuration)
 GLOBAL_PROTECT(config)
+
+/**
+ * THIS !!!SINGLE!!! PROC IS WHERE ANY FORM OF INIITIALIZATION THAT CAN'T BE PERFORMED IN SUBSYSTEMS OR WORLD/NEW IS DONE
+ * NOWHERE THE FUCK ELSE
+ * I DON'T CARE HOW MANY LAYERS OF DEBUG/PROFILE/TRACE WE HAVE, YOU JUST HAVE TO DEAL WITH THIS PROC EXISTING
+ * I'M NOT EVEN GOING TO TELL YOU WHERE IT'S CALLED FROM BECAUSE I'M DECLARING THAT FORBIDDEN KNOWLEDGE
+ * SO HELP ME GOD IF I FIND ABSTRACTION LAYERS OVER THIS!
+ */
+/world/proc/Genesis(tracy_initialized = FALSE)
+	RETURN_TYPE(/datum/controller/master)
+
+#ifdef USE_BYOND_TRACY
+#warn USE_BYOND_TRACY is enabled
+	if(!tracy_initialized)
+		init_byond_tracy()
+		Genesis(tracy_initialized = TRUE)
+		return
+#endif
+
+	// Init the debugger first so we can debug Master
+	init_debugger()
```

Comment:
are we sure we want to do this all the time

Reply:
This only happens if the env variable is set (inside the function it gets the env variable), if it's not set it doesn't load the debugger, also i think that's how it already works, you don't have to do anything special when you build to have the debugger in